packages/react-components/react-motion/library/src/hooks/useAnimateAtoms.ts

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import type { AnimationHandle, AtomMotion } from '../types';
+
+/**
+ * In test environments, this hook is used to delay the execution of a callback until the next render. This is necessary
+ * to ensure that the callback is not executed synchronously, which would cause the test to fail.
+ *
+ * @see https://github.com/microsoft/fluentui/issues/31701
+ */
+function useFinishDispatcher() {
+  const [count, setCount] = React.useState(0);
+  const callbackRef = React.useRef<() => void>();
+
+  React.useEffect(() => {
+    if (count > 0) {
+      callbackRef.current?.();
+    }
+  }, [count]);
+
+  return React.useCallback(callback => {
+    callbackRef.current = callback;
+    setCount(v => v + 1);
+  }, []);
+}
+
+function noop() {
+  /* no-op */
+}
+
+/**
+ * @internal
+ */
+export function useAnimateAtoms() {
+  // We don't want to use this hook in any environment other than test.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const dispatchFinishInTest = process.env.NODE_ENV === 'test' ? useFinishDispatcher() : noop;
```

Comment:
Why not wrap this entire thing in a conditional like you do where `dispatchFinishInTest()` is called? This will completely eliminate this in production builds.

Reply:
@spmonahan Not sure that I follow, do you mind to drop a suggestion?

Reply:
Where `dispatchFinishInTest` is called the entire block is wrapped in an environment check which should allow for all the wrapped code to be dead code eliminated.

<img width="421" alt="msedge_tNvk6DWNF0" src="https://github.com/microsoft/fluentui/assets/93940821/5b3cb742-cbb5-47a4-b905-a82756d95727">

I'm thinking we should do the same here so the code will be eliminated:

```ts
let dispatchFinishInTest;
if (process.env.NODE_ENV === 'test') {
  dispatchFinishInTest = useFinishDispatcher();
}

Reply:
Did a test and I think the ternary should work

<img width="1268" alt="msedge_ePSERhgxXn" src="https://github.com/microsoft/fluentui/assets/93940821/2c22b468-9565-44a0-84b9-b8eebdd19a9b">


Disregard my original comment :)